src/v/cluster/members_table.cc

```diff
@@ -145,7 +145,10 @@ members_table::apply(model::offset version, maintenance_mode_cmd cmd) {
     _version = model::revision_id(version());
 
     const auto target = _brokers.find(cmd.key);
-    if (target == _brokers.end()) {
+    if (
+      target == _brokers.end()
+      || target->second->get_membership_state()
+           == model::membership_state::removed) {
```

Comment:
this is preventing a decommissioned node from being placed into maintenance mode? if so, since we generally seem to allow this dual state, should we also prevent this case?

Reply:
It's preventing a _fully_ decommissioned/removed node from being put into maintenance mode, but it's okay to place a mode into maintenance mode if it's still draining. It isn't really necessary for the fix, but I can make the change if you feel like this is a path we need to preserve. FWIW this new behavior matches `dev`.

Reply:
got it. i think this is ok.

w.r.t. _It's preventing a fully decommissioned/removed node_ if `removed` corresponds to this state of being fully decommissioned, is there then a later point when the broker disappears from in-memory representations, corresponding to an even more fully removed state?

Reply:
I don't think so, ISTR we never erased from the nodes container until the refactor landed.